### PR TITLE
Make Multinomial robust against batches

### PR DESCRIPTION
**Pull request by lucianopaz**
At the moment, our `Multinomial` distribution mangles the `n` parameter's shape. This makes it very difficult to work with batches that have more than 2 dimensions. This PR addresses the problem and adds a test for it.

**Thank your for opening a PR!**

Depending on what your PR does, here are a few things you might want to address in the description:
+ [ ] what are the (breaking) changes that this PR makes?
+ [ ] important background, or details about the implementation
+ [X] are the changes—especially new features—covered by tests and docstrings?
+ [ ] consider adding/updating relevant example notebooks
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

### CODE REVIEW

**Comment by twiecki** [comment]
Seems like there are legit test errors.

**Comment by lucianopaz** [comment]
Yeah, I'll go through those tomorrow.

File: pymc3/distributions/multivariate.py [hunk 1]
```diff
@@ -597,14 +597,10 @@ def __init__(self, n, p, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         p = p / tt.sum(p, axis=-1, keepdims=True)
-        n = np.squeeze(n)  # works also if n is a tensor
```
Comment (twiecki) [inline pymc3/distributions/multivariate.py:600]: pretty simple fix, just remove some code :+1:

**Comment by twiecki** [comment]
Also needs a line in the release-notes.

**Comment by twiecki** [comment]
Thanks @lucianopaz!

**Review by AlexAndorra — COMMENTED** [review]
Thanks @lucianopaz ! Funny to see that _removing_ code fixed the issue 😅

**Pull request merged**